crates/parser/src/ast.rs

```diff
@@ -0,0 +1,125 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum Operator {
+    Equal,
+    NotEqual,
+    IsAbove,
+    IsAtLeast,
+    IsBelow,
+    IsAtMost,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Comparison {
+    pub op: Operator,
+    pub target_value: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "fn", content = "args")]
+pub enum AssertionKind {
+    IsUp {
+        node_name: String,
+        timeout: Option<u32>,
```

Comment:
nit: Using `Duration` might make sense and using a serde annotation to use a custom deserializer `fn`